File: DarkOrbit 10.0/Net/netty/requests/PetRequest.cs
using Ow.Utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ow.Net.netty.requests
{
    class PetRequest
    {
        public const short ID = 645;

        public const short LAUNCH = 0;
        public const short DEACTIVATE = 1;
        public const short TOGGLE_ACTIVATION = 2;
        public const short HOTKEY_GUARD_MODE = 3;
        public const short REPAIR_DESTROYED_PET = 4;
        public const short HOTKEY_REPAIR_SHIP = 5;

        public short petRequestType;

        public void readCommand(byte[] bytes)
        {
            var parser = new ByteParser(bytes);
            petRequestType = parser.readShort();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/clientCommands/PetRequest.java
package simulator.netty.clientCommands;

import java.io.DataInputStream;
import java.io.IOException;

import simulator.netty.ClientCommand;

/**
 Created by LEJYONER on 21/09/2017.
*/

public class PetRequest
        extends ClientCommand {

    public static final int ID = 9683;

    public static final short LAUNCH = 0;    
    public static final short DEACTIVATE = 1;    
    public static final short TOGGLE_ACTIVATION = 2;    
    public static final short HOTKEY_GUARD_MODE = 3;   
    public static final short REPAIR_DESTROYED_PET = 4;    
    public static final short HOTKEY_REPAIR_SHIP = 5;
     
    public short petRequestType = 0;
    
    public PetRequest(DataInputStream in) {
        super(in, ID);
    }

    public void readInternal() {
    	try {
	        this.petRequestType = in.readShort();
	        in.readShort();
	        in.readShort();
    	} catch (IOException e) {    		
    	}
    }
}